library/chacha20.c

```diff
@@ -210,6 +210,26 @@ int mbedtls_chacha20_starts(mbedtls_chacha20_context *ctx,
     return 0;
 }
 
+int mbedtls_chacha20_set_iv(mbedtls_chacha20_context *ctx,
+                            const unsigned char *nonce,
+                            size_t nlen,
+                            uint32_t counter)
+{
+    unsigned char iv[12] = { 0 };
```

Comment:
For the 12-byte case, it's a bit inefficient to make a copy of `nonce` as it is copied again by `mbedtls_chacha20_starts()`.